homeassistant/components/smartthings/smartapp.py

```diff
@@ -399,6 +402,25 @@ async def smartapp_install(hass: HomeAssistantType, req, resp, app):
     )
 
 
+def create_reauthorize_notification(hass: HomeAssistantType, entry: ConfigEntry):
+    """Create a notification to reauthorize the SmartApp."""
+    reauthorize_url = format_install_url(
+        entry.data[CONF_APP_ID], entry.data[CONF_LOCATION_ID]
+    )
+    hass.components.persistent_notification.async_create(
```

Comment:
Would it be possible to start a new config flow instead with an external step?

Reply:
No - the whole integration doesn't need to be setup again -- just this step to complete which sends a new `refresh token` to the webhook endpoint.

Reply:
We can abort after that step is completed. It's not super smooth, but a persistent notification is even more not meant for this. We want to use the config flow to update the config when the user needs to do something.

Reply:
I don't think that makes sense.  The config flow would require extensive changes as the external step is at the end.  We'd basically need an entirely new flow that has different steps, updates an existing entry, and aborts?  That's not a good user experience at all.  I don't see why we can't use a notification for this--it seems to be the optimal user experience.

Reply:
The user experience would be the same. See the samsungtv integration for an example:

https://github.com/home-assistant/core/blob/5bfc1f3d4d6e43bc02a127ac773d6cc45556bf03/homeassistant/components/samsungtv/media_player.py#L106-L113

https://github.com/home-assistant/core/blob/5bfc1f3d4d6e43bc02a127ac773d6cc45556bf03/homeassistant/components/samsungtv/config_flow.py#L163-L176

Reply:
I'm trying to implement this but here's what it's requiring:
- I need to version the config entries because ST isn't setting `unique_id`, which is needed for the config flow to replace an existing entry
- In the discovered entries, there isn't a way to specify a title (AFAIK) so all it says is "SmartThings" which isn't distinguishable.  Additionally the original integration is still there, which is very confusing and unclear what is wrong.
- At the completion of the flow, `async_remove_entry` is getting called on the old entry, which the ST integration was already using to clean up the ST account.  I now need to figure out how to introduce logic to not run this code if there's another in-flight entry.  This also breaks any custom named entries the user set for the integration.  

I don't think this is the right approach.

Reply:
This should indeed be done in a reauth with an external step. You can poll in the background if the user has clicked on allow. Note, you don't need to finish the flow with a create_entry. You can also update the existing entry and then abort it. That way you don't need to fight with `unload`. 

In 109 we're going to allow renaming config entries, so the titles should start making a bit more sense. 

For unique ID migration, we've been doing this: https://github.com/home-assistant/core/blob/188f3e35fdab484f368fcb6c238fed8392022f57/homeassistant/components/hue/__init__.py#L125-L129

Reply:
Oh and one more thing, you can specify a `flow_title` in your `strings.json/config` to specify a title with placeholders for a config flow. That way you know what it relates to: https://github.com/home-assistant/core/blob/dev/homeassistant/components/deconz/strings.json#L4

Reply:
Thanks I'll look into those things.  If I abort at the end after updating the config entry, is the user shown an "Abort" window?

Reply:
Yes, but you can control the translation text, so you just say: "Yo, auth updated, thank you! 🥂"